internal/route/node_filter.go

```diff
@@ -45,15 +98,30 @@ type IsCloseToGlobalMaxHeight struct {
 	maxBlocksBehind    uint64
 }
 
-func (f *IsCloseToGlobalMaxHeight) Apply(_ metadata.RequestMetadata, upstreamConfig *config.UpstreamConfig) bool {
+func (f *IsCloseToGlobalMaxHeight) Apply(
+	_ metadata.RequestMetadata,
+	upstreamConfig *config.UpstreamConfig,
+) bool {
+	upstreamStatus := f.healthCheckManager.GetUpstreamStatus(upstreamConfig.ID)
+	check := upstreamStatus.BlockHeightCheck
+
+	checkIsHealthy := check.GetError() == nil
+	if !checkIsHealthy {
+		zap.L().Debug("IsCloseToGlobalMaxHeight failed: most recent health check did not succeed.", zap.Error(check.GetError()))
+		return false
+	}
+
 	maxHeight := f.chainMetadataStore.GetGlobalMaxHeight()
+	upstreamHeight := check.GetBlockHeight()
+	isClose := upstreamHeight+f.maxBlocksBehind >= maxHeight
 
-	upstreamStatus := f.healthCheckManager.GetUpstreamStatus(upstreamConfig.ID)
+	if isClose {
+		return true
+	}
 
-	checkIsHealthy := upstreamStatus.BlockHeightCheck.GetError() == nil
-	isClose := upstreamStatus.BlockHeightCheck.GetBlockHeight()+f.maxBlocksBehind >= maxHeight
+	zap.L().Debug("Upstream too far behind global max height!", zap.Uint64("UpstreamHeight", upstreamHeight), zap.Uint64("MaxHeight", maxHeight))
```

Comment:
Should we add `upstreamConfig.ID` to these debug logs? To correlate which upstream the filter is acting upon.

Reply:
Yep! Was indeed already planning to do it as I browsed through the logs a bunch of times and had a bit of a hard time figuring out which message was about which upstream, especially when there were concurrent requests in-flight.